userspace/libscap/scap_procs.c

```diff
@@ -1714,24 +1670,24 @@ int32_t scap_check_suppressed(scap_t *handle, scap_evt *pevent, bool *suppressed
 	return SCAP_SUCCESS;
 }
 
-struct ppm_proclist_info *scap_procfs_get_threadlist(scap_t *handle)
+int32_t scap_procfs_get_threadlist(struct scap_engine_handle engine, struct ppm_proclist_info **procinfo_p)
 {
-	struct ppm_proclist_info *res = NULL;
 #ifdef __linux__
 	DIR *dir_p = NULL;
 	DIR *taskdir_p = NULL;
 	FILE *fp = NULL;
 	struct dirent *dir_entry_p;
 	char procdirname[SCAP_MAX_PATH_SIZE];
+	struct kmod_engine* kmod_engine = engine.m_handle;
```

Comment:
If i am not wrong, udig engine too has `scap_procfs_get_threadlist` in its vtable :)

Reply:
Done